### ISSUE TITLE
double.roundEven

### ISSUE THREAD

**Issue by Hixie** [dart-lang/sdk#46495]
It would be nice if `double` in Dart had a built-in way to do round-to-even, since that's the recommended mode in IEEE754. (Currently double.round does round-to-zero.)

The name "roundEven" is what OpenGL and libc use for this mode, FWIW.

**Comment by lrhn** [comment]
As far as I know, it's not directly supported in JavaScript.
You can probably use something like `2 * Math.round(n / 2)` (and zero if infinite or NaN, as normal when rounding to an integer value, keep them if `roundEvenAsDouble`).

**Comment by Hixie** [comment]
Yeah, the docs would probably have to mention that this is expensive in JavaScript environments.
